packages/node/src/lib/http-client.ts

```diff
@@ -0,0 +1,95 @@
+import { abortSignalAfterTimeout } from './abort'
+import { fetch as defaultFetch } from './fetch'
+
+/**
+ * This interface is meant to be compatible with different fetch implementations (node and browser).
+ * Using the ambient fetch type is not possible because the AbortSignal type is not compatible with node-fetch.
+ * @link https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API
+ */
+export interface HTTPFetchFn {
+  (url: string, requestInit: HTTPFetchRequest): Promise<HTTPResponse>
+}
+
+/**
+ * This interface is meant to be compatible with the Request interface.
+ * @link https://developer.mozilla.org/en-US/docs/Web/API/Request
+ */
+export interface HTTPFetchRequest {
+  headers?: Record<string, string>
+  body?: string
+  method?: string
+  signal?: any // AbortSignal type does not play nicely with node-fetch
+}
+
+/**
+ * This interface is meant to very minimally conform to the Response interface.
+ * @link https://developer.mozilla.org/en-US/docs/Web/API/Response
+ */
+export interface HTTPResponse {
+  ok: boolean
+  status: number
+  statusText: string
+}
+
+/**
+ * This interface is meant to be a generic interface for making HTTP requests.
+ * While it may overlap with fetch's Request interface, it is not coupled to it.
+ */
+export interface HTTPClientRequest {
+  /**
+   * URL to be used for the request
+   * @example 'https://api.segment.io/v1/batch'
+   */
+  url: string
+  /**
+   * HTTP method to be used for the request
+   * @example 'POST'
+   **/
+  method: string
+  /**
+   * Headers to be sent with the request
+   */
+  headers: Record<string, string>
+  /**
+   * JSON data to be sent with the request (will be stringified)
+   * @example { "batch": [ ... ]}
+   */
+  data: Record<string, any>
```

Comment:
It feels a little weird for an HTTP client to have `data` instead of `body`. (headers and query params are data too!)

How come we accept an object rather than a string/byte array here? I would expect the HTTP Client to only be concerned with making the HTTP request, not transforming the body.

Reply:
I can see the possibility of wanting to change the data before sending to the server and hence wanting to have the object and not parse from JSON.

But we have already a few places to do these operations, so I agree with this, let's keep the HTTP Client dumb to discourage having too much logic here

Reply:
`data` is used by `axios` to describe JSON data, rather than body, which is a string